src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -53,22 +53,29 @@ protected function render_button(Component\Component $component, RendererInterfa
 			$tpl->touchBlock("disabled");
 		}
 
-		$this->maybe_render_id($component, $tpl);
-
-		return $tpl->get();
+		$this->maybeRenderId($component, $tpl);
+        return $tpl->get();
 	}
 
-	protected function render_close($component) {
+
+	protected function renderClose($component) {
 		$tpl = $this->getTemplate("tpl.close.html", true, true);
 		// This is required as the rendering seems to only create any output at all
 		// if any var was set or block was touched.
 		$tpl->setVariable("FORCE_RENDERING", "");
-		$this->maybe_render_id($component, $tpl);
+		$this->maybeRenderId($component, $tpl);
 		return $tpl->get();
 	}
 
-	protected function maybe_render_id($component, $tpl) {
+	protected function maybeRenderId(Component\Component $component, $tpl) {
 		$id = $this->bindJavaScript($component);
+		if ($id === null) {
+			// No onload code available, check if the button is acting as triggerer
+			if ($component instanceof Component\Triggerer && count($component->getTriggeredSignals())) {
+				$id = $this->createId();
+				$this->triggerRegisteredSignals($component, $id);
+			}
+		}
```

Comment:
Maybe move this to some place where one could reuse this snippet. Maybe merge this somehow with bindJavascript, e.g. by adding an onload-code?